docs/aws/audit/kmsmonitoring/rules/secrets_manager_in_use_remediation.mdx

```diff
@@ -1,6 +1,256 @@
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and navigate to the AWS Key Management Service (KMS) dashboard.
+
+2. In the navigation pane, select "Customer managed keys". This will display a list of all the customer-managed keys that are currently available in your AWS account.
+
+3. For each key listed, click on its alias to open the key details page. Here, you can check the "Key users" section to see if AWS Secrets Manager is listed as a user. If it is not listed, then Secrets Manager is not in use for that particular key.
+
+4. Repeat the process for all the keys listed in the "Customer managed keys" section. If none of the keys have AWS Secrets Manager listed as a user, then Secrets Manager is not in use in KMS.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access KMS and Secrets Manager.
+
+2. Once AWS CLI is installed and configured, you can list all the keys in your account by running the following command:
+
+   ```
+   aws kms list-keys
+   ```
+   This command will return a list of all the keys in your account, including their key IDs and ARNs.
+
+3. To check if Secrets Manager is using these keys, you can list all the secrets in your account by running the following command:
+
+   ```
+   aws secretsmanager list-secrets
+   ```
+   This command will return a list of all the secrets in your account, including their names and ARNs.
+
+4. For each secret, you can describe the secret to see which KMS key it's using by running the following command:
+
+   ```
+   aws secretsmanager describe-secret --secret-id <secret-name>
+   ```
+   Replace `<secret-name>` with the name of the secret. This command will return information about the secret, including the KMS key that it's using. If the `KmsKeyId` field is not present or does not match any of the keys listed in step 2, then Secrets Manager is not using KMS.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you can start writing the script, you need to install the necessary Python libraries. AWS SDK for Python (Boto3) is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python that allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc. You can install it using pip:
+
+   ```bash
+   pip install boto3
+   ```
+
+2. Import the necessary libraries and create a session: In your Python script, you need to import the necessary libraries and create a session using your AWS credentials.
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>